### PR TITLE
Search by phone number in patient search view

### PR DESCRIPTION
**Pull request by vinaysshenoy**
Pivotal Story Link: https://www.pivotaltracker.com/story/show/167343626

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/patient/PatientRepository.kt [hunk 1]
```diff
@@ -93,6 +93,10 @@ class PatientRepository @Inject constructor(
         .doOnSubscribe { timingTracker.start(fetchPatientNameAnalytics) }
   }
 
+  fun searchByPhoneNumber(phoneNumber: String): Observable<List<PatientSearchResult>> {
+    TODO("not implemented")
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/patient/PatientRepository.kt]: This code path wont get executed in production because the source event never gets emitted from the UI. Will be implemented in another PR.

File: app/src/test/java/org/simple/clinic/searchresultsview/PatientSearchViewControllerTest.kt [hunk 2]
```diff
@@ -208,7 +210,129 @@ class PatientSearchViewControllerTest {
   private fun `params for search by parameters on register click`(): List<SearchPatientBy> {
     return listOf(
         SearchPatientBy.Name(searchText = patientName),
-        SearchPatientBy.PhoneNumber(searchText = "123456")
+        SearchPatientBy.PhoneNumber(searchText = phoneNumber)
     )
   }
+
+  @Test
+  fun `when searching patients by phone returns results, the results should be displayed`() {
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/searchresultsview/PatientSearchViewControllerTest.kt]: I plan to unify the tests for phone number and name search in a future PR.

Comment (ragunathjawahar) [reply app/src/test/java/org/simple/clinic/searchresultsview/PatientSearchViewControllerTest.kt]: Will that be a parameterized test?

Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/searchresultsview/PatientSearchViewControllerTest.kt]: I'm planning to make it one. I'll see what the final shape of the solution looks like when I get around to it.

Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/searchresultsview/PatientSearchViewControllerTest.kt]: @ragunathjawahar This unification was done in #735
